Author: gregor herrmann <gregoa@debian.org>
Bug: #482339
Description: use Sys::Hostname::Long in order to get FQDN for HELO
--- libmail-sendmail-perl.orig/Sendmail.pm
+++ libmail-sendmail-perl/Sendmail.pm
@@ -53,6 +53,7 @@
 use Socket;
 use Time::Local; # for automatic time zone detection
 use Sys::Hostname; # for use of hostname in HELO
+use Sys::Hostname::Long; # for use of hostname in HELO
 
 # use MIME::QuotedPrint if available and configured in %mailcfg
 eval("use MIME::QuotedPrint");
@@ -267,7 +268,7 @@
     }
 
     # get local hostname for polite HELO
-    $localhost = hostname() || 'localhost';
+    $localhost = hostname_long() || hostname() || 'localhost';
 
     foreach $server ( @{$mailcfg{'smtp'}} ) {
         # open socket needs to be inside this foreach loop on Linux,
@@ -766,6 +767,9 @@
 Now uses Sys::Hostname to get the hostname for HELO. (This may break the
 module on some very old Win32 Perls where Sys::Hostname was broken)
 
+(On Debian systems Sys::Hostname::Long is tried before Sys::Hostname in order
+get a fully qualified domain name.)
+
 Enable full session output for debugging
 
 See the F<Changes> file for the full history. If you don't have it
